docs/content/new-tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,221 @@
+
+# Note: This code will *not* run.
+@asset
+def hackernews_topstories():
+    results = []
+    for item_id in top_100_newstories:
+        item = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json").json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            print(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return df
+```
+
+You'll notice that the code is not syntactically correct because `top_100_newstories` is not a variable that the function can use. This is because the new `hackernews_topstories` SDA does not know that it is dependent on the `top_100_newstories` asset.
+
+Fixing that is easy! To add the asset to the function scope and have Dagster know that the `hackernews_topstories` depends on the `top_100_newstories` asset, add the name of the asset (`top_100_newstories`) as a parameter to the function definition of `hackernews_topstories`.
+
+```python
+import pandas as pd
+
+@asset
+def hackernews_topstories(top_100_newstories):
+    results = []
+    for item_id in top_100_newstories:
+        item = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json").json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            print(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return df
+```
+
+The code should now be free of syntax errors, and you can access the `top_100_newstories` asset. If you head over to the Dagster UI, you'll see that an asset graph is built for you. The asset graph is a visual representation of the assets and their dependencies.
+
+<Note>
+If this syntax is too magical for you, you can explicitly define an asset's dependencies a parameter called "inputs" in the @asset decorator. The explicit dependency syntax is also helpful when referencing assets from other asset groups. TODO: Add link to this.
+</Note>
+
+
+### Context and logging
+
+When defining an asset, you can optionally create a `context` parameter for the function. The `context` argument contains information about the asset, such as the asset's name, the run ID, and the logger.
+
+Using the `context` argument gives you access to features. With the `context.log` object, you can print text that will appear with the job's run in the Dagster UI. With this, you can log the progress of your asset like below.
+
+```python
+import pandas as pd
+
+@asset
+def hackernews_topstories(context, top_100_newstories):
+    results = []
+    for item_id in top_100_newstories:
+        item = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json").json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            context.log.info(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return df
+```
+
+For more information about loggers, such as the different types of logging, you can read the [logging documentation](/concepts/logging/loggers).
```

Comment:
```suggestion
For more information about loggers, such as the different types of logging, refer to the [logging documentation](/concepts/logging/loggers).
```

⛏️ Sort of a nit, sorry. IMO using 'understood you' for most references to other docs reads better.